tests/misc/t12383.nim

```diff
@@ -0,0 +1,7 @@
+var b = "some string"
+
+doAssert type(b[0..b.len-1]) is string
+doAssert typeof(b[0..b.len-1]) is string
```

Comment:
```nim
var b = "some string"
doAssert typeof(b[0..b.len-1], typeOfIter) is char
doAssert typeof(b[0..b.len-1], typeOfProc) is string
```
Should be added to test cases too which works before.

Reply:
Isn't typeOfIter the default? Then the old behavior is correct.

Reply:
Yeah, I agree

Reply:
yeah I misunderstand the `typeof ` design, thought it like other languages do.